compiler-plugin/src/main/java/io/ballerina/stdlib/tcp/compiler/TcpServiceValidator.java

```diff
@@ -63,12 +64,14 @@ public void validate() {
                     ExpressionNode expressionNode = returnStatementNode.expression().get();
 
                     if (expressionNode instanceof ExplicitNewExpressionNode) { // handle return new HelloService();
-                        TypeReferenceTypeSymbol typeReferenceTypeSymbol = (TypeReferenceTypeSymbol) ctx.semanticModel()
-                                .type(expressionNode).get();
-                        ClassSymbol classSymbol = (ClassSymbol) typeReferenceTypeSymbol.typeDescriptor();
-                        TcpConnectionServiceValidator tcpConnectionServiceValidator =
-                                new TcpConnectionServiceValidator(ctx, classSymbol);
-                        tcpConnectionServiceValidator.validate();
+                        TypeSymbol symbol = ctx.semanticModel().type(expressionNode).get();
+                        if (symbol instanceof TypeReferenceTypeSymbol) {
```

Comment:
Instead of `instanceof` check, better to use symbol kind to check the symbol type. 
```suggestion
                        if (symbol.typeKind() == TypeDescKind.TYPE_REFERENCE) {
```